src/odemis/gui/cont/tabs.py

```diff
@@ -6411,7 +6411,7 @@ def add_axis(axisname, comp, label=None):
             if ((not almost_equal(spec_switch_xpos, spec_switch_xmd_deactive)) and
                     (not almost_equal(spec_switch_xpos, spec_switch_xmd_active))):
                 self._spec_switch_f = main.spec_switch.moveAbs({"x": spec_switch_xmd_deactive})
-                self._spec_switch_f.add_done_callback(self._on_specswitch_button_done)
```

Comment:
Ok, this avoids touching the GUI while the tab is hidden, and anyway, once the LIGHT IN alignment mode is selected, `_adjust_spec_switch_button_state()` should do all the needed part.
Can you entirely remove the line, instead of just commenting it out. Maybe add a comment indicating we don't update the GUI explicitly as the tab is hidden, and that messes up with the progress bar position.